src/components/Modal/BaseModal.tsx

```diff
@@ -45,6 +47,15 @@ function ModalComponent({type, ...props}: ModalComponentProps) {
             />
         );
     }
+    if (type && PARTIAL_REANIMATED_MODAL_TYPES.includes(type) && shouldUseReanimatedModal) {
+        return (
+            <ReanimatedModal
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...(props as ReanimatedModalProps)}
+                type={type}
+            />
+        );
+    }
```

Comment:
NAB: What if we remove `PARTIAL_REANIMATED_MODAL_TYPES` and just rely on `shouldUseReanimatedModal`?

This way we would end up with just one if:
```ts
    if ((type && REANIMATED_MODAL_TYPES.includes(type)) || shouldUseReanimatedModal)
```